### ISSUE TITLE
Warn when `# noqa: <code>` contains an invalid code

### ISSUE THREAD

**Issue by martinlehoux** [astral-sh/ruff#3339]
With the following snippet, I can ignore the issue in 3 ways, but the 4th fails to ignore, and it's not consistant:
- Add `# noqa: N815` on the line :heavy_check_mark: 
- Add `# noqa: mixed-case-variable-in-class-scope` on the line :heavy_check_mark: 
- Add `# ruff: noqa: N815` on the first line of the file :heavy_check_mark:
- Add `# ruff: noqa: mixed-case-variable-in-class-scope` on the first line of the file :stop_sign: 

```py
from typing import TypedDict


class Request(TypedDict):
    queryString: str -> Variable `queryString` in class scope should not be mixedCaseRuff(N815)
```

**Comment by charliermarsh** [comment]
The second way is actually kind of a bug but it's consistent with Flake8 IIUC -- `# noqa: mixed-case-variable-in-class-scope` ignores _all_ errors in the file.

**Comment by martinlehoux** [comment]
:open_mouth: Damn that could lead me to unexpected ignores, I thought it was normal behavior. I have `# noqa: hardcoded-password-string` or `# noqa: ambiguous-unicode-character-string` all over the place

**Comment by charliermarsh** [comment]
I thought we warned about that, I'll have to double-check. (I see stuff like `# noqa: F401` often, which Flake8 treats as "ignore all rules".)

We don't currently support using the human-readable names for ignores, we need to finalize the names first.

**Comment by charliermarsh** [comment]
Wait sorry sorry.

**Comment by charliermarsh** [comment]
I misspoke. Let me clarify.

**Comment by charliermarsh** [comment]
`# noqa: F401` only ignores `F401` on a given line.

`# flake8: noqa: F401` on its own line ignores _all_ errors in a file.

**Comment by charliermarsh** [comment]
`# noqa: mixed-case-variable-in-class-scope` ignores all errors on a given line, erroneously (in both Ruff and Flake8).

**Comment by charliermarsh** [comment]
If you do `# ruff: noqa: mixed-case-variable-in-class-scope`, we do properly warn that `mixed-case-variable-in-class-scope` is an unknown code.

**Comment by charliermarsh** [comment]
We should probably warn that `# noqa: mixed-case-variable-in-class-scope` is an unknown code.
